src/makeRobot.js

```diff
@@ -38,7 +38,68 @@
  * @return {object}
  */
 function makeRobot(name, wheels, version) {
-  // write code here
+  const robot = {
+    name: name,
+    wheels: wheels,
+    version: version,
+    coords: {
+      x: 0,
+      y: 0,
+    },
+
+    get info() {
+      return `name: ${this.name}, chip version: ${this.version}, wheels: ${this.wheels}`;
+    },
+
+    get location() {
+      return `${this.name}: x=${this.coords.x}, y=${this.coords.y}`;
+    },
+
+    goLeft: function(k = 1) {
+      if (Math.abs(k) !== k) {
+        return this;
+      }
+      this.coords.x = this.coords.x - k;
+
+      return this;
+    },
```

Comment:
Hard solution. You can simplify this. 
```suggestion
    goLeft: function(k = 1) {
      if (k > 0) {
        this.coords.x = this.coords.x - k;
      }
      
      return this;
    },
```
And in next method